app/src/main/java/at/roteskreuz/stopcorona/screens/debug/exposure_notifications/DebugExposureNotificationsFragment.kt

```diff
@@ -33,9 +35,31 @@ class DebugExposureNotificationsFragment : BaseFragment(R.layout.debug_contact_t
 
         exposureNotificationsSettingsButton.setOnClickListener { viewModel.jumpToSystemSettings() }
 
-        exposureNotificationsUploadTemporaryExposureKeysGreenButton.setOnClickListener { viewModel.uploadKeys(WarningType.REVOKE) }
-        exposureNotificationsUploadTemporaryExposureKeysRedButton.setOnClickListener { viewModel.uploadKeys(WarningType.RED) }
-        exposureNotificationsUploadTemporaryExposureKeysYellowButton.setOnClickListener { viewModel.uploadKeys(WarningType.YELLOW) }
+        val uploadKeylistener = View.OnClickListener(){button ->
+            val tan = exposureNotificationsTanEditText.text.toString()
+            if (tan.isNullOrBlank()){
+                activity?.let { Toast.makeText(activity,"please add TAN", Toast.LENGTH_SHORT)}
+                exposureNotificationsTanEditText.error = "please provide TAN"
+                return@OnClickListener
+            }else {
+                exposureNotificationsTanEditText.error = null
+            }
+            val warningType = when (button) {
+                exposureNotificationsUploadTemporaryExposureKeysGreenButton -> WarningType.REVOKE
+                exposureNotificationsUploadTemporaryExposureKeysRedButton -> WarningType.RED
+                exposureNotificationsUploadTemporaryExposureKeysYellowButton -> WarningType.YELLOW
+                else -> throw IllegalArgumentException()
+            }
+
+            viewModel.uploadKeys(warningType, tan)
+        }
+        exposureNotificationsUploadTemporaryExposureKeysGreenButton.setOnClickListener(uploadKeylistener)
+        exposureNotificationsUploadTemporaryExposureKeysRedButton.setOnClickListener(uploadKeylistener)
+        exposureNotificationsUploadTemporaryExposureKeysYellowButton.setOnClickListener(uploadKeylistener)
+
+        exposureNotificationsPhoneNumberEditText.setText("+4915157401235")
```

Comment:
This should not go into production!

Reply:
If it is your private phone, I think you will notice implication very fast 😄